File: scripts/uninstall.mjs
#!/usr/bin/env node

// Claude Toolkit Uninstall - Zero dependencies, Node built-ins only
import { execSync, spawnSync } from 'node:child_process';
import { createInterface } from 'node:readline';
import { platform } from 'node:os';
import { existsSync } from 'node:fs';
import { join } from 'node:path';

const isWin = platform() === 'win32';

function log(msg = '') {
  process.stdout.write(msg + '\n');
}

function commandExists(cmd) {
  try {
    execSync(isWin ? `where ${cmd}` : `command -v ${cmd}`, { stdio: 'pipe' });
    return true;
  } catch {
    return false;
  }
}

function checkMcp(name) {
  try {
    const out = execSync('claude mcp list', { encoding: 'utf8', stdio: 'pipe' });
    return out.toLowerCase().includes(name.toLowerCase());
  } catch {
    return false;
  }
}

function checkPlugin(name) {
  try {
    const out = execSync('claude plugin list', { encoding: 'utf8', stdio: 'pipe' });
    return out.toLowerCase().includes(name.toLowerCase());
  } catch {
    return false;
  }
}

function run(cmd, opts = {}) {
  try {
    execSync(cmd, { stdio: 'inherit', ...opts });
    return true;
  } catch {
    log(`  Warning: command failed: ${cmd}`);
    return false;
  }
}

function ask(rl, question) {
  return new Promise((resolve) => rl.question(question, resolve));
}

// ---------------------------------------------------------------------------
// Tool definitions (uninstall only)
// ---------------------------------------------------------------------------

const TOOLS = {
  context7: {
    name: 'Context7',
    check: () => checkMcp('context7'),
    uninstall: () => run('claude mcp remove --scope user context7'),
  },

  serena: {
    name: 'Serena',
    check: () => checkMcp('serena'),
    uninstall: () => run('claude mcp remove --scope user serena'),
  },

  beads: {
    name: 'Beads (plugin)',
    check: () => checkPlugin('beads'),
    uninstall: () => {
      run('claude plugin uninstall beads');
      run('claude plugin marketplace remove beads-marketplace');
    },
    note: 'The bd CLI is not removed. Uninstall it separately (brew uninstall beads / npm uninstall -g @beads/bd).',
  },

  beadsui: {
    name: 'beads-ui',
    check: () => commandExists('bdui'),
    uninstall: () => run('npm uninstall -g beads-ui'),
  },

  bmad: {
    name: 'BMAD-METHOD',
    check: () => existsSync(join(process.cwd(), '_bmad')),
    uninstall: () => {
      log('  Launching BMAD uninstaller...');
      const result = spawnSync('npx bmad-method uninstall', {
        stdio: 'inherit',
        shell: true,
      });
      return result.status === 0;
    },
    note: 'BMAD is per-project. Run this from the project directory.',
  },

};

// ---------------------------------------------------------------------------
// Main
// ---------------------------------------------------------------------------

async function main() {
  log();
  log('  Claude Toolkit Uninstall');
  log('  ========================');
  log();

  if (!commandExists('claude')) {
    log('Error: `claude` CLI not found.');
    process.exit(1);
  }

  // Check what's installed
  log('Checking installed tools...');
  log();

  const installed = [];
  for (const [id, tool] of Object.entries(TOOLS)) {
    let found = false;
    try {
      found = await tool.check();
    } catch {
      found = false;
    }

    const icon = found ? '\x1b[32m✓\x1b[0m' : '\x1b[90m·\x1b[0m';
    log(`  ${icon} ${id.padEnd(10)} ${tool.name}`);
    if (found) installed.push(id);
  }

  log();

  if (installed.length === 0) {
    log('Nothing to uninstall.');
    return;
  }

  const rl = createInterface({ input: process.stdin, output: process.stdout });

  log(`Installed: ${installed.join(', ')}`);
  const answer = await ask(rl, 'Remove (space-separated IDs, "all", or "q" to quit): ');
  rl.close();

  const trimmed = answer.trim().toLowerCase();
  if (!trimmed || trimmed === 'q') {
    log('Cancelled.');
    process.exit(0);
  }

  const selected = trimmed === 'all' ? installed : trimmed.split(/\s+/);

  for (const id of selected) {
    const tool = TOOLS[id];
    if (!tool) {
      log(`\nUnknown tool: ${id}`);
      continue;
    }

    log(`\nRemoving ${tool.name}...`);
    await tool.uninstall();
    log(`✓ ${tool.name} removed`);
    if (tool.note) log(`  Note: ${tool.note}`);
  }

  log();
  log('  Uninstall complete!');
  log('  Restart Claude Code for changes to take effect.');
  log();
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: scripts/uninstall.py
#!/usr/bin/env python3
"""Claude Toolkit Uninstall — remove installed tools. Stdlib only."""

import os
import subprocess
import sys

sys.path.insert(0, os.path.dirname(__file__))
from lib import (
    BOLD, DIM, GREEN, RESET,
    check_mcp, check_plugin, command_exists, log, run,
)

# ── Tool definitions (uninstall only) ─────────────────────────────────────────

TOOLS = {
    "atlas": {
        "name": "Atlas",
        "check": lambda: check_plugin("atlas"),
        "uninstall": lambda: run("claude plugin uninstall famdeck-atlas"),
        "note": "Per-project .claude/atlas.yaml files and ~/.claude/atlas/ registry are preserved.",
    },
    "relay": {
        "name": "Relay",
        "check": lambda: check_plugin("relay"),
        "uninstall": lambda: run("claude plugin uninstall famdeck-relay"),
        "note": "Per-project .claude/relay.yaml files are preserved.",
    },
    "context7": {
        "name": "Context7",
        "check": lambda: check_mcp("context7"),
        "uninstall": lambda: run("claude mcp remove --scope user context7"),
    },
    "serena": {
        "name": "Serena",
        "check": lambda: check_mcp("serena"),
        "uninstall": lambda: run("claude mcp remove --scope user serena"),
    },
    "beads": {
        "name": "Beads (plugin)",
        "check": lambda: check_plugin("beads"),
        "uninstall": lambda: (
            run("claude plugin uninstall beads"),
            run("claude plugin marketplace remove beads-marketplace"),
        ),
        "note": "The bd CLI is not removed. Uninstall it separately.",
    },
    "beadsui": {
        "name": "beads-ui",
        "check": lambda: command_exists("bdui"),
        "uninstall": lambda: run("npm uninstall -g beads-ui"),
    },
    "mail": {
        "name": "Agent Mail",
        "check": lambda: check_mcp("agent-mail"),
        "uninstall": lambda: run("claude mcp remove --scope user agent-mail"),
        "note": "Server repo at ~/.mcp_agent_mail/ not removed. Delete manually if needed.",
    },
    "bmad": {
        "name": "BMAD-METHOD",
        "check": lambda: os.path.isdir(os.path.join(os.getcwd(), "_bmad")),
        "uninstall": lambda: subprocess.run("npx bmad-method uninstall", shell=True).returncode == 0,
        "note": "BMAD is per-project. Run this from the project directory.",
    },
}

# ── Main ──────────────────────────────────────────────────────────────────────


def main():
    log()
    log(f"  {BOLD}Claude Toolkit Uninstall{RESET}")
    log(f"  {'=' * 24}")
    log()

    if not command_exists("claude"):
        log("Error: `claude` CLI not found.")
        sys.exit(1)

    log("Checking installed tools...")
    log()

    installed = []
    for id_, t in TOOLS.items():
        try:
            found = t["check"]()
        except Exception:
            found = False

        icon = f"{GREEN}✓{RESET}" if found else f"{DIM}·{RESET}"
        log(f"  {icon} {id_:<10} {t['name']}")
        if found:
            installed.append(id_)

    log()

    if not installed:
        log("Nothing to uninstall.")
        return

    log(f"Installed: {', '.join(installed)}")
    try:
        answer = input('Remove (space-separated IDs, "all", or "q" to quit): ').strip().lower()
    except (EOFError, KeyboardInterrupt):
        log("\nCancelled.")
        sys.exit(0)

    if not answer or answer == "q":
        log("Cancelled.")
        sys.exit(0)

    selected = installed if answer == "all" else answer.split()

    for id_ in selected:
        t = TOOLS.get(id_)
        if not t:
            log(f"\nUnknown tool: {id_}")
            continue

        log(f"\nRemoving {t['name']}...")
        t["uninstall"]()
        log(f"✓ {t['name']} removed")
        if t.get("note"):
            log(f"  Note: {t['note']}")

    log()
    log(f"  {BOLD}Uninstall complete!{RESET}")
    log("  Restart Claude Code for changes to take effect.")
    log()


if __name__ == "__main__":
    main()
